### PR TITLE
8277807: Increase default initial concurrent refinement threshold

### PR DESCRIPTION
**Pull request by kimbarrett**
Please review this change to the default initial concurrent refinement
threshold.  The new value takes into account options that should provide a
starting point that is closer to the long-term range.

Testing:
mach5 tier1
manually verified logged initial value matched expectations.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8277807](https://bugs.openjdk.java.net/browse/JDK-8277807): Increase default initial concurrent refinement threshold

### Reviewers
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to 0adb6a5a685f2df9c9bd3cd6a6cac36aa8be5496
 * [Stefan Johansson](https://openjdk.java.net/census#sjohanss) (@kstefanj - **Reviewer**) ⚠️ Review applies to 0adb6a5a685f2df9c9bd3cd6a6cac36aa8be5496

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6549/head:pull/6549` \
`$ git checkout pull/6549`

Update a local copy of the PR: \
`$ git checkout pull/6549` \
`$ git pull https://git.openjdk.java.net/jdk pull/6549/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6549`

View PR using the GUI difftool: \
`$ git pr show -t 6549`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6549.diff">https://git.openjdk.java.net/jdk/pull/6549.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/g1/g1ConcurrentRefine.cpp [hunk 1]
```diff
@@ -251,13 +251,19 @@ static size_t calc_min_yellow_zone_size() {
...
 static size_t calc_init_green_zone() {
-  size_t green = G1ConcRefinementGreenZone;
-  const char* name = "G1ConcRefinementGreenZone";
+  size_t green;
   if (FLAG_IS_DEFAULT(G1ConcRefinementGreenZone)) {
-    green = ParallelGCThreads;
-    name = "ParallelGCThreads";
+    // rate_1 is a relatively conservative guess at the rate for pause-time
+    // card refinement by one thread.
+    const double rate_1 = 200.0; // cards/ms/thread
```
Comment (tschatzl) [inline src/hotspot/share/gc/g1/g1ConcurrentRefine.cpp:258]: I would prefer if this magic number were placed somewhere more standing out than in some static method in a cpp file. I thought at least put it into the .hpp file as part of the `G1ConcurrentRefine" class. However there are no guidelines for that for g1 code where or how to collect these magic constants I'm kind of good with that too.

**Review by tschatzl — APPROVED** [review]
Lgtm.

File: src/hotspot/share/gc/g1/g1ConcurrentRefine.cpp (hunk 1 above)
Comment (kstefanj) [reply src/hotspot/share/gc/g1/g1ConcurrentRefine.cpp:258]: I agree with Thomas. I think this should live in `G1ConcurrentRefine`, but if you don't agree I'm ok with leaving as is.

**Review by kstefanj — APPROVED** [review]

File: src/hotspot/share/gc/g1/g1ConcurrentRefine.cpp (hunk 1 above)
Comment (kimbarrett) [reply src/hotspot/share/gc/g1/g1ConcurrentRefine.cpp]: Moving it into the class would have required moving the referencing static
function into the class, &etc. I've given it a more informative name and
description, and moved it to file scope to make it more prominent.

**Comment by kimbarrett** [comment]
Thanks @kstefanj and @tschatzl for reviews.

**Pull request closed without merging**